core/typedef.go

```diff
@@ -472,6 +480,23 @@ func (typeDef *TypeDef) WithObjectConstructor(fn *Function) (*TypeDef, error) {
 	return typeDef, nil
 }
 
+func (typeDef *TypeDef) WithEnum(name, desc string) *TypeDef {
+	typeDef = typeDef.WithKind(TypeDefKindEnum)
+	typeDef.AsEnum = dagql.NonNull(NewEnumTypeDef(name, desc))
+	return typeDef
+}
+
+func (typeDef *TypeDef) WithEnumValue(name, desc string) (*TypeDef, error) {
+	if !typeDef.AsEnum.Valid {
+		return nil, fmt.Errorf("cannot add value to non-enum type: %s", typeDef.Kind)
+	}
+
+	typeDef = typeDef.Clone()
+	typeDef.AsEnum.Value.Values = append(typeDef.AsEnum.Value.Values, NewEnumValueTypeDef(name, desc))
```

Comment:
Need to error if it's a duplicate. Otherwise it overwrites a previous one, which can be confusing. Doesn't have to be here, just that it's validated by the API at some point.